tests/ga/test_signature_validation.py

```
@@ -0,0 +1,69 @@
+        with self.assertRaises(ExtensionDisallowedError):
+            validate_signature(self.null_ext_zip_path, invalid_signature)
+
+    def test_should_raise_error_if_package_is_invalid(self):
+        # This is the null extension zip package with one byte modified
+        modified_ext = os.path.join(data_dir, "signing/NullExtensionModified.zip")
+        with self.assertRaises(ExtensionDisallowedError):
+            validate_signature(modified_ext, self.null_ext_signature)
+
+    def test_should_raise_error_on_invalid_root_certificate(self):
+        invalid_root_cert_path = os.path.join(data_dir, "signing/invalid_root_cert.pem")
+        with patch("azurelinuxagent.ga.signature_validation.conf.get_microsoft_root_certificate_path", return_value=invalid_root_cert_path):
+            with self.assertRaises(ExtensionDisallowedError):
+                validate_signature(self.null_ext_zip_path, self.null_ext_signature)
```

can we test an expired certificate?

---

but the whole point is that for now we are allowing expired certificates... this test should pass

---

Ah, I see, I misunderstood your comment earlier. I added a test for an expired Microsoft root certificate that was not originally used to sign the extension. This case should fail since it's essentially another invalid cert test case.

However, test_should_validate_signature_successfully() already covers the scenario you're asking about. The intermediate certificate that signed NullExtension.zip has already expired, and this test would fail if we didn’t set the -no_check_time parameter. From what I understand, we can’t modify the expiration date of the root cert used to sign the extension, so testing with the expired intermediate cert seemed like a reasonable alternative.

---

We need to test both expired intermediate and root certs.

---

Added a positive test case for expired root certificate (note that I re-signed the test extension with an expired root cert I generated myself, not a valid Microsoft root cert)

---

this is a critical scenario, can you work with the signing authority to  get an extension (or any ZIP) signed with an expired cert?

---

Added these cases to test_signature_validation_sudo.py